front/src/common/useSearchUsers.ts

```diff
@@ -15,47 +17,34 @@ export type User = {
 
 export default function useSearchUsers() {
   const [searchTerm, setSearchTerm] = useState('');
-  const [searchedUsers, setSearchedUsers] = useState<User[]>([]);
-
   const debouncedSearchTerm = useDebounce(searchTerm, 150);
-  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
 
-  const searchUser = useCallback(async (searchQuery: string) => {
-    try {
-      const results = (await postSearch({
-        searchPayload: {
-          object: 'user',
-          query: ['search', ['name'], searchQuery],
-        },
-        pageSize: 101,
-      }).unwrap()) as SearchResultItemUser[];
-
-      return results.map((result) => ({
+  const { data: searchedUsersData } = osrdEditoastApi.endpoints.postSearch.useQuery(
```

Comment:
We can use `currentData` here to make it so the results are reset when `debouncedSearchTerm` changes (instead of persisting the old value until we receive a new response).

Reply:
(BTW - all of the above is a nit, feel free to ignore/resolve.)